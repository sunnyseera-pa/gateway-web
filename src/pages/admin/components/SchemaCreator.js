/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Dropdown from '../../../components/Dropdown';
import Input from '../../../components/Input';
import Checkbox from '../../../components/Checkbox';
import Icon from '../../../components/Icon';

import { ReactComponent as LockIcon } from '../../../images/icons/lock.svg';
import { ReactComponent as PlusIcon } from '../../../images/icons/plus.svg';
import { ReactComponent as EyeIcon } from '../../../images/icons/eye.svg';
import LayoutBox from '../../../components/LayoutBox';
import IconButton from '../../../components/IconButton/IconButton';

const inputTypes = ['textInput', 'textInputCustom'];

const SchemaCreator = ({ type }) => {
    const [schema, setSchema] = React.useState([
        {
            title: 'Section',
            id: 'people',
            subsections: [
                {
                    title: 'Sub section',
                    id: 'primaryapplicant',
                    fields: [
                        {
                            type: 'textInput',
                            label: 'Full name',
                            required: 1,
                            lockedQuestion: 0,
                            defaultQuestion: 1,
                        },
                    ],
                },
            ],
        },
    ]);

    const handleAddField = React.useCallback(
        (schemaIndex, subsectionIndex) => {
            const updatedSchema = _.update(schema, `[${schemaIndex}].subsections[${subsectionIndex}].fields`, fields => {
                return fields.concat({
                    type: 'textInput',
                    label: '[Label]',
                    required: 1,
                    lockedQuestion: 0,
                    defaultQuestion: 1,
                });
            });

            setSchema([...updatedSchema]);
        },
        [schema]
    );

    console.log('updatedSchema', schema);

    return (
        <LayoutBox m={4}>
            {schema.map(({ title, subsections }, schemaIndex) => (
                <div>
                    <h2>{title}</h2>
                    {subsections.map(({ title, fields }, subsectionIndex) => (
                        <div>
                            <h3>
                                {title}{' '}
                                <IconButton
                                    icon={<Icon color='purple500' svg={<PlusIcon />} />}
                                    onClick={() => handleAddField(schemaIndex, subsectionIndex)}
                                />
                            </h3>
                            {fields.map(({ label, type, lockedQuestion, defaultQuestion }) => (
                                <LayoutBox display='flex' alignItems='flex-end' mb={4}>
                                    {type === 'textInput' && <Input label={label} mr={6} minWidth='400px' />}

                                    <Dropdown options={inputTypes} value={type} mr={2} />
                                    <LayoutBox display='flex' flexDirection='column'>
                                        <Icon svg={<LockIcon />} size='xl' mb={1} />
                                        <Checkbox mb={0} checked={lockedQuestion} />
                                    </LayoutBox>
                                    <LayoutBox display='flex' flexDirection='column'>
                                        <Icon svg={<EyeIcon />} size='xl' mb={1} fill='purple500' />
                                        <Checkbox mb={0} checked={defaultQuestion} />
                                    </LayoutBox>
                                </LayoutBox>
                            ))}
                        </div>
                    ))}
                </div>
            ))}
        </LayoutBox>
    );
};

SchemaCreator.propTypes = {
    type: PropTypes.oneOf(['DAR']),
};

SchemaCreator.defaultProps = {
    type: 'DAR',
};

export default SchemaCreator;
