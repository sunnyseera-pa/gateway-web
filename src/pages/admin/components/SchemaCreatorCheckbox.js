import React from 'react';
import Checkbox from '../../../components/Checkbox';
import Icon from '../../../components/Icon';
import LayoutBox from '../../../components/LayoutBox';

const SchemaCreatorCheckbox = ({ field, questionSetIndex, fieldIndex, checked, onChange, icon }) => {
    const handleOnChange = React.useCallback(
        data => {
            onChange(field, questionSetIndex, fieldIndex, data);
        },
        [field, questionSetIndex, fieldIndex]
    );

    return (
        <LayoutBox display='flex' flexDirection='column' alignItems='center' mr={2}>
            <Icon svg={icon} mb={1} fill='purple500' size='xl' />
            <Checkbox mb={0} checked={checked} onChange={handleOnChange} />
        </LayoutBox>
    );
};

export default SchemaCreatorCheckbox;
