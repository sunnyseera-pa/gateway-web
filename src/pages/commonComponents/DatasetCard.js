import React, { Fragment, useState } from 'react';
import { OverlayTrigger, Tooltip, Row, Button, Accordion } from 'react-bootstrap';
import { isEmpty } from 'lodash';
import SLA from '../commonComponents/sla/SLA';
import TimeDuration from './timeDuration/TimeDuration';
import DatasetOnboardingHelper from './../../utils/DatasetOnboardingHelper.util';
import CommentItem from '../dashboard/DataAccessRequests/CommentItem/CommentItem.js';
import StatusDisplay from './StatusDisplay';
import SVGIcon from '../../images/SVGIcon';
import moment from 'moment';

import '../commonComponents/DatasetCard.scss';

let completionList = {
	Summary: 'summary',
	Documentation: 'documentation',
	Coverage: 'coverage',
	Provenance: 'provenance',
	Accessibility: 'accessibility',
	Enrichment: 'enrichmentAndLinkage',
	Observations: 'observations',
	'Structural Meta Data': 'structural',
};

export const DatasetCard = props => {
	let {
		id,
		publisher,
		title,
		version,
		datasetStatus,
		completion = {},
		timeStamps = {},
		rejectionText,
		rejectionAuthor,
		listOfVersions,
	} = props;
	const [flagClosed, setFlagClosed] = useState(true);

	return (
		<Row key={`dataset_card_${title}`} onClick={() => (window.location.href = `/dataset-onboarding/${id}`)}>
			<div className='col-md-12'>
				<div className='layoutCard mb-0'>
					<div className='datasetHeader mb-0 mt-2'>
						<div className='datasetHeader-title'>
							<h1>{title}</h1>
						</div>
						<div className='datasetHeader-status'>
							{datasetStatus === 'draft' ? (
								<TimeDuration text={`${DatasetOnboardingHelper.calculateTimeDifference(timeStamps.created)} days since start`} />
							) : (
								''
							)}
							{datasetStatus === 'inReview' ? (
								<TimeDuration text={`${DatasetOnboardingHelper.calculateTimeDifference(timeStamps.submitted)} days since submission`} />
							) : (
								''
							)}
							<SLA
								classProperty={DatasetOnboardingHelper.datasetStatusColours[datasetStatus]}
								text={DatasetOnboardingHelper.datasetSLAText[datasetStatus]}
							/>

							{datasetStatus === 'draft' && listOfVersions.find(version => version.activeflag === 'active') ? (
								<>
									&nbsp;&nbsp;
									<SLA
										classProperty={DatasetOnboardingHelper.datasetStatusColours['active']}
										text={DatasetOnboardingHelper.datasetSLAText['active']}
									/>
								</>
							) : (
								''
							)}
						</div>
					</div>
					<div className='dataset-completion-wheels'>
						{!isEmpty(completion)
							? Object.keys(completionList).map(key => (
									<OverlayTrigger
										key={key}
										placement='top'
										overlay={
											<Tooltip id={`tooltip-top`}>
												{key}: {completion[completionList[key]]}
											</Tooltip>
										}>
										<div>
											<StatusDisplay section={key} status={completion[completionList[key]]} />
										</div>
									</OverlayTrigger>
							  ))
							: ''}
					</div>

					<div className='body'>
						<Fragment>
							<div className='box'>Publisher</div>
							<div className='box'>{publisher}</div>
							<div className='box version-list'>Version</div>

							{listOfVersions.length > 1 ? (
								<div className='box'>
									<Accordion defaultActiveKey='1' style={{ width: '100%' }}>
										<Accordion.Toggle
											as={Button}
											variant='link'
											eventKey='0'
											onClick={e => {
												e.stopPropagation();
												setFlagClosed(!flagClosed);
											}}
											data-testid='accordion-toggle'
											style={{ width: '100%', padding: '0px', border: '0px' }}
											className='version-list'>
											<div className='version-list'>
												{version}
												{datasetStatus === 'draft' ? ' (Draft)' : ''}
												{datasetStatus === 'inReview' ? ' (Pending)' : ''}
												<SVGIcon
													name='chevronbottom'
													fill={'#475da7'}
													style={{ width: '18px', height: '18px', paddingLeft: '4px' }}
													className={flagClosed === true ? 'svg-24' : 'svg-24 flipSVG'}
												/>
											</div>
										</Accordion.Toggle>
										<Accordion.Collapse eventKey='0' style={{ paddingRight: '20px' }}>
											<Fragment>
												{listOfVersions.map(datasetVersion => (
													<>
														{datasetVersion.datasetVersion !== version ? (
															<a
																href='javascript:void(0)'
																className='version-list'
																onClick={e => {
																	e.stopPropagation();
																	window.location.href = `/dataset-onboarding/${datasetVersion._id}`;
																}}>
																{datasetVersion.datasetVersion}
																{datasetVersion.activeflag === 'draft' ? ' (Draft)' : ''}
																{datasetVersion.activeflag === 'active' ? ' (Live)' : ''}
																{datasetVersion.activeflag === 'rejected' ? ' (Rejected)' : ''}
																{datasetVersion.activeflag === 'inReview' ? ' (Pending)' : ''}
															</a>
														) : (
															''
														)}
													</>
												))}
											</Fragment>
										</Accordion.Collapse>
									</Accordion>
								</div>
							) : (
								<div className='box'>
									{version}
									{datasetStatus === 'draft' ? ' (Draft)' : ''}
								</div>
							)}
							{(() => {
								console.log(timeStamps.updated);
							})()}
							<div className='box'>Last activity</div>
							<div className='box'>{!isEmpty(timeStamps.updated) ? moment(timeStamps.updated).format('D MMMM YYYY HH:mm') : '-'}</div>
						</Fragment>
					</div>
					{datasetStatus === 'rejected' ? (
						<CommentItem
							text={rejectionText}
							title={'Reason for rejection'}
							subtitle={rejectionAuthor || 'HDR Admin'}
							decisionDate={moment(timeStamps.rejected).format('D MMMM YYYY HH:mm')}
						/>
					) : (
						''
					)}
				</div>
			</div>
		</Row>
	);
};

export default DatasetCard;
