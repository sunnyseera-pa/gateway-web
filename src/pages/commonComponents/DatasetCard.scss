@import '../../css/typography';

.dataset-completion-wheels {
	display: flex;
	flex-flow: row;
	max-width: 145px;
	width: 100%;
	align-items: center;
	justify-content: space-between;
	margin: 0 25px;
}

.version-list {
	display: flex;
	flex-flow: wrap;
	align-items: flex-start !important;
	word-break: break-word;
	position: relative;
	font-family: museo-sans-rounded;
	font-weight: 500;
	color: $gray-800;
	font-size: 14px;
}
