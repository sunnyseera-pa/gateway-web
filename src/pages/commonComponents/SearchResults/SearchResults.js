import React from 'react';
import { Pagination, Row } from 'react-bootstrap';
import Loading from '../Loading';
import NoResults from '../NoResults';
import RelatedObject from '../relatedObject/RelatedObject';
import { DEFAULT_PROPS_SEARCH_RESULTS, PROP_TYPES_SEARCH_RESULTS } from './SearchResults.propTypes';

const SearchResults = ({
	results,
	type,
	count,
	search,
	data,
	sort,
	pageNumber,
	onPagination,
	maxResults,
	updateOnFilterBadge,
	isLoading,
	totalPages,
	errorMessage,
}) => (
	<>
		<div style={{ marginTop: '8px' }}>
			{!isLoading && (
				<>
					{sort && (
						<Row>
							<div className='text-right save-dropdown'>{sort}</div>
						</Row>
					)}
					{count <= 0 && (!errorMessage ? <NoResults type={type} searchString={search} /> : errorMessage({ search, type }))}
					{!results &&
						count > 0 &&
						data.map(item => {
							return (
								<RelatedObject key={item.id} data={item} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />
							);
						})}
					{results && count > 0 && results(data)}
					{count > maxResults && (
						<Pagination>
							{new Array(Math.ceil(totalPages)).fill().map((value, i) => (
								<Pagination.Item key={i} active={i === pageNumber} onClick={() => onPagination(type, i * maxResults)}>
									{i + 1}
								</Pagination.Item>
							))}
						</Pagination>
					)}
				</>
			)}
			{!!isLoading && <Loading data-testid='loader' />}
		</div>
	</>
);

SearchResults.propTypes = PROP_TYPES_SEARCH_RESULTS;

SearchResults.defaultProps = DEFAULT_PROPS_SEARCH_RESULTS;

export default SearchResults;
