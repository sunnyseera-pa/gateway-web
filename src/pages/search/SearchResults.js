import { Pagination } from 'react-bootstrap';

const typeMapper = {
	Datasets: 'dataset',
	Tools: 'tool',
	Projects: 'project',
	Papers: 'paper',
	People: 'person',
	Courses: 'course',
	Collections: 'collection',
};

const SearchResults = ({
	results,
	type,
	count,
	search,
	data,
	filters,
	sort,
	pageNumber,
	onPagination,
	maxResult,
	updateOnFilterBadge,
	onAdvancedSearchClick,
	isLoading,
}) => {
	<Container>
		<Row>
			{filters && (
				<Col sm={12} md={12} lg={3} className='mt-1 mb-5'>
					<div className='saved-filterHolder'>{filters}</div>
				</Col>
			)}
			{!filters && <Col sm={12} md={12} lg={3} />}
			<div className='advanced-search-link-container'>
				<CDStar fill='#f98e2b' height='20' width='20' />
				<a
					href='javascript:void(0)'
					className='textUnderline gray800-14 cursorPointer'
					onClick={() => {
						googleAnalytics.recordEvent('Datasets', 'User clicked advanced search link', 'Advanced search modal opened');
						if (onAdvancedSearchClick) onAdvancedSearchClick();
					}}>
					Advanced Search
				</a>
			</div>
			{!isLoading && (
				<Col sm={12} md={12} lg={9} className='mt-1 mb-5'>
					<Row>
						<div className='text-right save-dropdown'>{sort}</div>
					</Row>
					{count <= 0 && <NoResults type={type} search={search} />}
					{!results &&
						count > 0 &&
						data.map(item => {
							return (
								<RelatedObject key={item.id} data={item} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />
							);
						})}
					{results && results(data)}
					{count >
						maxResult(
							<Pagination>
								{new Array(Math.ceil(count / maxResult)).fill().map((value, i) => {
									<Pagination.Item
										key={i}
										active={i === pageNumber / maxResult}
										onClick={() => onPagination(typeMapper[type], i * maxResult)}>
										{i}
									</Pagination.Item>;
								})}
							</Pagination>
						)}
				</Col>
			)}
			{!!isResultsLoading && (
				<Col style={{ marginTop: '30px' }} sm={12} md={12} lg={9}>
					<Loading />
				</Col>
			)}
		</Row>
	</Container>;
};

export default SearchResults;
