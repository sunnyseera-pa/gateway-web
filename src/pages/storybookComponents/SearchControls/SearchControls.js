import React from 'react';
import { Formik } from 'formik';
import SearchControlsForm from './SearchControlsForm';

const SearchControls = ({ onSubmit, isLoading, sortProps, ...outerProps }) => {
	if (isLoading) return null;

	return (
		<Formik initialValues={{ search: '', sortBy: sortProps ? sortProps.value : '' }} onSubmit={onSubmit}>
			<SearchControlsForm sortProps={sortProps} {...outerProps} />
		</Formik>
	);
};

export default SearchControls;
