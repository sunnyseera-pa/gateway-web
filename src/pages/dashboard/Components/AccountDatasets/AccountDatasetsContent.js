import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import pluralize from 'pluralize';
import DatasetCard from '../../../commonComponents/DatasetCard';
import SearchResults from '../../../commonComponents/SearchResults';
import Icon from '../../../../components/Icon';
import SearchControls from '../../../../components/SearchControls';
import '../../Dashboard.scss';
import MessageNotFound from '../../../commonComponents/MessageNotFound';

const AccountDatasetsContent = ({ data = [], onSubmit, isLoading, isFetched, status, params, team, count }) => {
	const [searchValue, setSearchValue] = useState();
	const history = useHistory();

	const { t } = useTranslation();
	const handleChange = React.useCallback(value => {
		setSearchValue(value);
	}, []);

	const getDatasetPath = id => {
		return `/account/datasets/${id}`;
	};

	const handleActivityLogClick = id => {
		history.push(getDatasetPath(id));
	};

	const hasActivityHistory = React.useCallback(
		dataset => {
			return dataset.listOfVersions.length > 0 && team === 'admin';
		},
		[team]
	);

	const getDatasetCardProps = dataset => {
		const datasetCardProps = {};

		if (dataset.activeflag === 'inReview' && hasActivityHistory(dataset)) {
			datasetCardProps.slaProps = {
				icon: <Icon name='eye' role='button' onClick={() => handleActivityLogClick(dataset.pid)} mr={1} />,
			};

			datasetCardProps.path = getDatasetPath(dataset.pid);
		} else if (dataset.activeflag === 'rejected') {
			datasetCardProps.rejectionText = dataset.applicationStatusDesc;
			datasetCardProps.rejectionAuthor = dataset.applicationStatusAuthor;
		}

		return datasetCardProps;
	};

	const { search, sortBy, sortDirection, maxResults } = params;

	return (
		<>
			{isFetched && (
				<SearchControls
					type={t(`dataset.${status}`)}
					mt={3}
					onSubmit={onSubmit}
					inputProps={{
						onChange: handleChange,
						mt: 2,
						value: search,
					}}
					sortProps={{
						direction: sortDirection,
						value: sortBy,
						options: ['alphabetic', 'latest', 'recentlyadded', 'metadata', 'popularity'],
						mt: 2,
						allowDirection: true,
						minWidth: '300px',
					}}
				/>
			)}

			<SearchResults
				data={data}
				errorMessage={({ type }) => <MessageNotFound word={pluralize(type)} />}
				results={data =>
					data.map(dataset => (
						<DatasetCard
							key={dataset._id}
							id={dataset._id}
							title={dataset.name}
							publisher={status !== 'archive' && dataset.datasetv2.summary.publisher.name}
							version={dataset.datasetVersion}
							isDraft={true}
							datasetStatus={dataset.activeflag}
							timeStamps={dataset.timestamps}
							completion={dataset.percentageCompleted}
							listOfVersions={dataset.listOfVersions}
							{...getDatasetCardProps(dataset)}
						/>
					))
				}
				count={count}
				maxResults={maxResults}
				type='dataset'
				isLoading={isLoading}
				search={searchValue}
			/>
		</>
	);
};

AccountDatasetsContent.defaultProps = {
	params: {
		search: '',
		sortBy: 'latest',
		sortDirection: 'desc',
	},
};

export default AccountDatasetsContent;
