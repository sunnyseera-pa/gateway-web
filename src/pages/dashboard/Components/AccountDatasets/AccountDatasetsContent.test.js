import '@testing-library/jest-dom/extend-expect';
import { act, render, waitFor } from '@testing-library/react';
import React from 'react';
import * as SearchControls from '../../../../components/SearchControls';
import { mockGetPublisher } from '../../../../services/dataset-onboarding/mockMsw';
import * as DatasetCard from '../../../commonComponents/DatasetCard';
import * as SearchResults from '../../../commonComponents/SearchResults';
import AccountDatasetsContent from './AccountDatasetsContent';

const mockOnSubmit = jest.fn();
const mockHistoryPush = jest.fn();

const searchResultsSpy = jest.spyOn(SearchResults, 'default');
const searchControlsSpy = jest.spyOn(SearchControls, 'default');
const datasetCardSpy = jest.spyOn(DatasetCard, 'default');

jest.mock('../../../../components/Icon', () => ({ onClick }) => (
	<span onClick={onClick} className='icon-mock'>
		Icon
	</span>
));

jest.mock('../../../commonComponents/relatedObject/RelatedObject', () => <div />);

jest.mock('react-router', () => ({
	...jest.requireActual('react-router'),
	useHistory: () => ({
		push: mockHistoryPush,
	}),
}));

const props = {
	data: [],
	onSubmit: mockOnSubmit,
	isLoading: true,
	isFetched: false,
	status: 'inReview',
	team: 'admin',
};

let wrapper;

describe('Given the AccountDatasetsContent component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<AccountDatasetsContent {...props} />, {
				wrapper: Providers,
			});
		});

		it('The should call SearchResults with the correct arguments', () => {
			expect(searchResultsSpy.mock.calls[0][0]).toMatchObject({
				count: 0,
				data: [],
				isLoading: true,
			});
		});

		it('The should not call SearchControls with the correct arguments', () => {
			expect(searchControlsSpy).not.toHaveBeenCalled();
		});

		describe('And has data', () => {
			beforeAll(() => {
				jest.clearAllMocks();

				wrapper = render(
					<AccountDatasetsContent {...props} data={mockGetPublisher.data.results.listOfDatasets} isFetched={true} isLoading={false} />,
					{
						wrapper: Providers,
					}
				);
			});

			it('Then matches the previous snapshot', () => {
				expect(wrapper.container).toMatchSnapshot();
			});

			it('Then handles input change', async () => {
				const input = wrapper.container.querySelector('input[name="search"]');

				await fireEvent.change(input, {
					target: {
						value: 'dataset',
					},
				});

				await waitFor(() => expect(input.value).toEqual('dataset'));
			});

			it('The should call SearchResults with the correct arguments', () => {
				expect(searchResultsSpy.mock.calls[0][0]).toEqual({
					count: 19,
					data: mockGetPublisher.data.results.listOfDatasets,
					isLoading: false,
					search: undefined,
					results: expect.any(Function),
					type: 'dataset',
				});
			});

			it('The should call Datasetcard with the correct arguments', () => {
				const {
					_id: id,
					name: title,
					datasetv2: {
						summary: {
							publisher: { name: publisher },
						},
					},
					datasetVersion: version,
					activeflag: datasetStatus,
					timestamps: timeStamps,
					percentageCompleted: completion,
					listOfVersions,
				} = mockGetPublisher.data.results.listOfDatasets[0];

				expect(datasetCardSpy.mock.calls[0][0]).toEqual({
					id,
					title,
					publisher,
					version,
					datasetStatus,
					timeStamps,
					completion,
					isDraft: true,
					listOfVersions,
				});
			});

			describe('And status is inReview', () => {
				beforeAll(() => {
					jest.clearAllMocks();

					wrapper = render(
						<AccountDatasetsContent
							{...props}
							data={mockGetPublisher.data.results.listOfDatasets}
							isFetched={true}
							isLoading={false}
							status='inReview'
						/>,
						{
							wrapper: Providers,
						}
					);
				});

				it('Then should change history onClick', async () => {
					const sla = wrapper.container.querySelectorAll('.sla-icons .icon-mock')[0];

					await fireEvent.click(sla, '1234');

					await waitFor(() => expect(mockHistoryPush).toHaveBeenCalled());
				});

				it('Then should call Datasetcard with the correct arguments', () => {
					const {
						_id: id,
						name: title,
						datasetv2: {
							summary: {
								publisher: { name: publisher },
							},
						},
						datasetVersion: version,
						activeflag: datasetStatus,
						timestamps: timeStamps,
						percentageCompleted: completion,
						listOfVersions,
					} = mockGetPublisher.data.results.listOfDatasets[1];

					expect(datasetCardSpy.mock.calls[1][0]).toEqual({
						id,
						title,
						publisher,
						version,
						datasetStatus,
						timeStamps,
						completion,
						isDraft: true,
						listOfVersions,
						path: '/account/datasets/0a048419-0796-46fb-ad7d-91e650a6c742',
						slaProps: expect.any(Object),
					});
				});
			});

			describe('And status is rejected', () => {
				beforeAll(() => {
					jest.clearAllMocks();

					wrapper = render(
						<AccountDatasetsContent
							{...props}
							data={mockGetPublisher.data.results.listOfDatasets}
							isFetched={true}
							isLoading={false}
							status='rejected'
						/>,
						{
							wrapper: Providers,
						}
					);
				});

				it('The should call Datasetcard with the correct arguments', () => {
					const {
						_id: id,
						name: title,
						datasetv2: {
							summary: {
								publisher: { name: publisher },
							},
						},
						datasetVersion: version,
						activeflag: datasetStatus,
						timestamps: timeStamps,
						percentageCompleted: completion,
						listOfVersions,
					} = mockGetPublisher.data.results.listOfDatasets[2];

					expect(datasetCardSpy.mock.calls[2][0]).toEqual({
						id,
						title,
						publisher,
						version,
						datasetStatus,
						timeStamps,
						completion,
						isDraft: true,
						listOfVersions,
						rejectionAuthor: 'Callum Reekie',
						rejectionText: 'SDg',
					});
				});
			});

			describe('And results are sorted', () => {
				beforeAll(() => {
					const input = wrapper.container.querySelector('button');

					fireEvent.click(input);
				});

				it('Then submits with the correct values', async () => {
					const link = wrapper.container.querySelectorAll('a')[1];

					fireEvent.click(link);

					await waitFor(() => expect(mockOnSubmit.mock.calls[0][0]).toEqual({ search: '', sortBy: 'latest', sortDirection: 'desc' }));
				});

				it('Then submits with the correct sort direction', async () => {
					const link = wrapper.container.querySelector('.btn-link');

					fireEvent.click(link);

					await waitFor(() => expect(mockOnSubmit.mock.calls[1][0]).toEqual({ search: '', sortBy: 'latest', sortDirection: 'asc' }));
				});
			});
		});
	});
});
