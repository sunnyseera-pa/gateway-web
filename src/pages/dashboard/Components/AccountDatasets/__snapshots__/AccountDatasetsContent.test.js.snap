// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccountDatasetsContent component When it is rendered And has data Then matches the previous snapshot 1`] = `
<div>
  <div
    class="css-1fe7u86 row"
  >
    <div
      class="col-lg-6"
    >
      <div
        class="ui-Input ui-SearchInput css-nxqzw4 css-107dhd5-Input input-group"
      >
        <div
          class="css-j8o1xl-Input input-group-prepend"
        >
          <span
            class="icon-mock"
          >
            Icon
          </span>
        </div>
        <input
          class="form-control"
          name="search"
          placeholder="Search within pending approval"
          value=""
        />
      </div>
    </div>
    <div
      class="col-lg-2"
    />
    <div
      class="d-flex justify-content-end col-lg-4"
    >
      <div
        class="ui-SortDropdown css-qy5j8w css-16k9oib-SortDropdown"
      >
        <div
          class="ui-Dropdown css-1sivg29 css-1quv89l-SortDropdown dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-primary"
            type="button"
          >
            Latest
          </button>
        </div>
        <button
          class="css-11u46hl-SortDropdown btn btn-link"
          ml="2"
          type="button"
        >
          <span
            class="ui-SortDropdown__desc"
          >
            <span
              class="icon-mock"
            >
              Icon
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    style="margin-top: 8px;"
  />
</div>
`;
